app/sign-in/page.tsx

```diff
@@ -1,13 +1,6 @@
-import { auth } from '@/auth'
 import { LoginButton } from '@/components/login-button'
-import { redirect } from 'next/navigation'
 
 export default async function SignInPage() {
-  const session = await auth()
-  // redirect to home if user is already logged in
-  if (session?.user) {
-    redirect('/')
-  }
```

Comment:
Moved this logic to Middleware

Reply:
So this is actually needed to be here, since NextAuth.js does not run Middleware on its own pages, to avoid an infinite redirect loop.